asn1crypto/x509.py

```diff
@@ -947,8 +947,10 @@ def native(self):
                 for type_val in rdn:
                     field_name = type_val['type']
                     if field_name in self._native:
-                        self._native[field_name] = [self._native[field_name]]
-                        self._native[field_name].append(type_val['value'])
+                        existing = self._native[field_name]
+                        if not isinstance(existing, list):
+                            existing = self._native[field_name] = [existing]
+                        existing.append(type_val['value'])
                     else:
```

Comment:
We should try to add some tests for this if we can.

Your changes will work if the first instance of a field is a list, however if the first instance is not a list, but the second is, I believe we will still get nested lists.

I think we'll need to add an `if` statement to handle if the `type_val['value']` is a list, and instead call `existing.extend(type_val['value'])`.